sdk/device_update/src/client.rs

```diff
@@ -134,42 +144,48 @@ impl DeviceUpdateClient {
 fn extract_endpoint(url: &Url) -> Result<String> {
     let endpoint = url
         .host_str()
-        .ok_or(Error::DomainParse)?
-        .splitn(2, '.') // FIXME: replace with split_once() when it is in stable
-        .last()
-        .ok_or(Error::DomainParse)?;
+        .ok_or_else(|| {
+            Error::with_message(ErrorKind::DataConversion, || {
+                format!("could not get device update domain. url: {url}")
+            })
+        })?
+        .split_once('.')
```

Comment:
I also noticed the `FIXME: use split_once`
Might as well do it as it is in stable now :)